extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -377,8 +377,14 @@ export class PerNotebookServerInstance implements IServerInstance {
 		return childProcess;
 	}
 
-	private getEnvWithConfigPaths(env: {}): any {
-		let newEnv: { [key: string]: string } = Object.assign({}, env);
+	private getEnvWithConfigPaths(env: { [key: string]: string }): any {
+		// Merge the current process.env object with the specified object so that we can pass information to notebooks
+		// Variables in the env object will overwrite the ones in process.env object
+		let newEnv: { [key: string]: string } = Object.assign({}, process.env);
```

Comment:
I'm worries this will conflict with our implementation in terms of Python lookup paths. Can you sync with @corivera  and @chlafreniere regarding this and make sure that Python install scenarios (Python already on path, etc.) are not broken.

Reply:
@kevcunnane , whatever was there will not get overwritten, the change will only take the new environment variables, the PATH and other environment variables set by notebook extension will be preserved.

Reply:
So there are funky variables like PYTHON_PATH that can get set, right @corivera ? Having seen this regress like 3 times already, I'm very wary which just means "let's test on a bunch of machines where we've seen regressions already". Cory you should remember this through the pain of multiple fixes right? 😉

Reply:
Cory explained me the context and I made changes so that we only take verify specific environment variables to avoid regression. @kevcunnane  and @corivera , does it look good now?